packages/headless/src/app/product-recommendation-engine/product-recommendation-engine.ts

```diff
@@ -74,7 +74,7 @@ export function buildProductRecommendationEngine(
 
   const thunkArguments = {
```

Comment:
could that variable be typed?

Reply:
I added the type `SearchThunkExtraArgument` and the legacy key `searchAPIClient` as suggested by Sami